### PR TITLE
Remove requirement for interop container

### PR DESCRIPTION
**Pull request by Xerkus**
zend-servicemanager ^3.4 allows plugin managers to use non-interop psr container.

This PR:
- bumps zend-servicemanager to ^3.4
- removes check in view helper plugin manager factory requiring psr container to also implement interop interface
- removes support for 2.* releases of zend-servicemanager

### CODE REVIEW

**Comment by Xerkus** [comment]
@weierophinney this PR enables usage of zend-view with containers other than zend-servicemanager but it have to drop no longer maintained major version of dependency. Needs your sign-off.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,8 @@ All notable changes to this project will be documented in this file, in reverse
 
 ### Added
 
-- Nothing.
+- [#65](https://github.com/zendframework/zend-expressive-zendviewrenderer/pull/65)
+  enables usage of non-interop psr containers with view helper plugin manager
```
Comment (weierophinney) [inline CHANGELOG.md]: "non-interop psr containers" feels like it is enabling usage of non-standard containers. I'd rephrase to something like "usage of any PSR-11 container with the view helper plugin manager".

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,7 +19,8 @@ All notable changes to this project will be documented in this file, in reverse
 
 ### Removed
 
-- Nothing.
+- [#65](https://github.com/zendframework/zend-expressive-zendviewrenderer/pull/65)
+  removes support for 2.* series of zendframework/zend-servicemanager
```
Comment (weierophinney) [inline CHANGELOG.md]:
```suggestion
  removes support for v2 releases of zendframework/zend-servicemanager
```

**Review by weierophinney — CHANGES_REQUESTED** [review]
I think this makes sense. However, I'd bump to a new major version, as we're dropping support for an entire major version of releases of one of our dependencies.

Once we do, we'll need to make sure the skeleton application installs the new major version by default when this renderer is selected. (That can be a bugfix of minor release, however, as the skeleton is not a dependency of a project, but a starting point.)

Only changes I've suggested are for the changelog itself.
